### PR TITLE
Streamline password reset flow

### LINKED ISSUE #3094: Streamline password reset flow
When resetting their password, users are sent an email containing a (pretty long) token and the link to the password reset form. They need to copy the token and insert it in the password reset view.

It would be a better user experience if the token was included in the url to the password reset form (as a GET parameter) and would be auto-filled. In that case users only need to click the link from the email and then enter their new password.

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
This PR puts the token used to reset your password (send by mail) directly into the url as a query parameter. The frontend the extracts this param so that the user only has to enter the new password.

### URL of deployed dev instance (used for testing):
- [https://streamlinepasswordrestflow.webknossos.xyz](https://streamlinepasswordrestflow.webknossos.xyz)

### Steps to test:
- create a new user (with one of your real email addresses) and activate him
- log out and request to reset the password of the new user
- go to your email account and use the link to reset your password
- you do not need to enter the token yourself anymore and the resetting should still work correctly

### Issues:
- fixes #3094

------
- ~~~[ ] Updated [changelog](../blob/master/CHANGELOG.md#unreleased)~~~
- ~~~[ ] Updated [migration guide](../blob/master/MIGRATIONS.md#unreleased) if applicable~~~
- ~~~[ ] Needs datastore update after deployment~~~
- [x] Ready for review

### CODE REVIEW

**Comment by normanrz** [comment]
I think it would be more elegant if you could take the token from the router instead of parsing the url directly.
See https://github.com/scalableminds/webknossos/blob/master/app/assets/javascripts/admin/auth/registration_view.js#L12 and https://github.com/scalableminds/webknossos/blob/master/app/assets/javascripts/router.js#L344

File: app/assets/javascripts/messages.js [hunk 1]
```diff
@@ -123,6 +123,8 @@ In order to restore the current window, a reload is necessary.`,
   "auth.reset_new_password": "Please input your new password!",
   "auth.reset_new_password2": "Please repeat your new password!",
   "auth.reset_token": "Please input the token!",
+  "auth.reset_token_not_supplied":
+    "There was no token was found in the URL. Check your E-Mails to get the correct URL.",
```
Comment (philippotto) [inline app/assets/javascripts/messages.js]: `There was no token found in the URL. ...`

Comment (MichaelBuessemeyer) [reply app/assets/javascripts/messages.js]: :smile:

**Comment by philippotto** [comment]
> Maybe redirect / instead?

You did that now, @MichaelBuessemeyer, right? It's always useful to leave a comment when you think this PR should be "reviewed" again (or accepted), as we don't keep track of all incoming commits for PRs.

@normanrz Should be ready to accept, right?

**Pull request merged**